File: C-5-SDT/OOP_and_Python_Programming/M3.5/Y_Range_sum_query.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/Y

n, q = input().split()
n, q = int(n), int(q)

arr = [int(x) for x in input().split()]
prefix_sum = [0] * (n + 1) # [0, 0, 0, 0, 0, 0, 0] n+1 size with 0s, and 1-index list
# print(prefix_sum)
for i in range(1, n + 1):
    prefix_sum[i] = prefix_sum[i - 1] + arr[i - 1]

for _ in range(q):
    l, r = input().split()
    l, r = int(l), int(r)
    result = prefix_sum[r] - prefix_sum[l - 1]
    print(result)

File: C-3-DS/M2.5/Y_Range_sum_query.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/Y

// Next module provide optimization of this code
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int N, Q;
    cin >> N >> Q;

    vector<int> a(N + 1);
    for (int i = 1; i <= N; i++)
        cin >> a[i];

    while (Q--)
    {
        int left, right;
        cin >> left >> right;

        // Brute force solution
        int sum = 0;
        for (int j = left; j <= right; j++) // get TLE - O(N)
            sum += a[j];

        cout << sum << endl;
    }
    return 0;
}